packages/fx-core/src/plugins/resource/bot/functionsHostedBot/deployMgr.ts

```diff
@@ -40,11 +40,17 @@ export class FuncHostedDeployMgr {
     }
   }
 
-  public async needsToRedeploy(ignoreRules: string[]): Promise<boolean> {
+  public async needsToRedeploy(): Promise<boolean> {
```

Comment:
Is it possible to always run `build` stage before checking changes?
In case certain ignored file impacts the build output, e.g. some specific settings in `.npmrc`.

Reply:
If we run `build` stage first, the output files in `dist/` will be updated. So the project will always be deployed.